Forbidden API Usage in OntoUploader

When I attempt to build master branch I get the following
```
[INFO] --- forbiddenapis:1.8:check (default) @ eskg ---
[INFO] Scanning for classes to check...
[INFO] Reading bundled API signatures: jdk-unsafe
[INFO] Reading bundled API signatures: jdk-deprecated
[INFO] Reading bundled API signatures: jdk-system-out
[INFO] Loading classes to check...
[INFO] Scanning for API signatures and dependencies...
[ERROR] Forbidden method invocation: java.io.InputStreamReader#<init>(java.io.InputStream) [Uses default charset]
[ERROR]   in org.esipfed.eskg.portalservice.OntoUploader (OntoUploader.java:27)
[ERROR] Forbidden method invocation: java.lang.Throwable#printStackTrace() [Eclipse auto-generated stubs; exceptions should be correctly bubbled up and handled accordingly]
[ERROR]   in org.esipfed.eskg.portalservice.OntoUploader (OntoUploader.java:33)
[ERROR] Forbidden method invocation: java.lang.String#getBytes() [Uses default charset]
[ERROR]   in org.esipfed.eskg.portalservice.OntoUploader (OntoUploader.java:59)
[ERROR] Forbidden method invocation: java.io.InputStreamReader#<init>(java.io.InputStream) [Uses default charset]
[ERROR]   in org.esipfed.eskg.portalservice.OntoUploader (OntoUploader.java:73)
[ERROR] Forbidden method invocation: java.lang.Throwable#printStackTrace() [Eclipse auto-generated stubs; exceptions should be correctly bubbled up and handled accordingly]
[ERROR]   in org.esipfed.eskg.portalservice.OntoUploader (OntoUploader.java:81)
[ERROR] Forbidden method invocation: java.io.InputStreamReader#<init>(java.io.InputStream) [Uses default charset]
[ERROR]   in org.esipfed.eskg.portalservice.OntoUploader (OntoUploader.java:129)
[ERROR] Forbidden method invocation: java.lang.Throwable#printStackTrace() [Eclipse auto-generated stubs; exceptions should be correctly bubbled up and handled accordingly]
[ERROR]   in org.esipfed.eskg.portalservice.OntoUploader (OntoUploader.java:137)
[ERROR] Forbidden field access: java.lang.System#out [prints to System.out; should only be used for debugging, not in production code]
[ERROR]   in org.esipfed.eskg.portalservice.OntoUploader (OntoUploader.java:145)
[ERROR] Forbidden field access: java.lang.System#out [prints to System.out; should only be used for debugging, not in production code]
[ERROR]   in org.esipfed.eskg.portalservice.OntoUploader (OntoUploader.java:150)
[ERROR] Scanned 5 (and 292 related) class file(s) for forbidden API invocations (in 0.10s), 9 error(s).
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 1.623 s
[INFO] Finished at: 2017-03-29T18:59:04-07:00
[INFO] Final Memory: 25M/255M
[INFO] ------------------------------------------------------------------------
```
PR coming soon.

--- a/doc/eclipse-formatter.xml
+++ b/doc/eclipse-formatter.xml
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <profiles version="11">
-<profile kind="CodeFormatterProfile" name="IFC-to-RDF Convertor" version="11">
+<profile kind="CodeFormatterProfile" name="ESKG" version="11">
 <setting id="org.eclipse.jdt.core.formatter.insert_space_before_closing_paren_in_if" value="do not insert"/>
 <setting id="org.eclipse.jdt.core.formatter.insert_space_after_colon_in_assert" value="insert"/>
 <setting id="org.eclipse.jdt.core.formatter.brace_position_for_enum_constant" value="end_of_line"/>
@@ -21,7 +21,7 @@
 <setting id="org.eclipse.jdt.core.formatter.format_guardian_clause_on_one_line" value="false"/>
 <setting id="org.eclipse.jdt.core.formatter.comment.insert_new_line_before_root_tags" value="insert"/>
 <setting id="org.eclipse.jdt.core.formatter.insert_space_after_colon_in_for" value="insert"/>
-<setting id="org.eclipse.jdt.core.formatter.tabulation.size" value="4"/>
+<setting id="org.eclipse.jdt.core.formatter.tabulation.size" value="2"/>
 <setting id="org.eclipse.jdt.core.formatter.insert_space_after_opening_angle_bracket_in_type_parameters" value="do not insert"/>
 <setting id="org.eclipse.jdt.core.formatter.blank_lines_before_imports" value="1"/>
 <setting id="org.eclipse.jdt.core.formatter.insert_space_after_colon_in_case" value="insert"/>

--- a/pom.xml
+++ b/pom.xml
@@ -308,6 +308,7 @@
   </profiles>
 
   <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <!-- Dependencies -->
 
     <junit.version>4.7</junit.version>

--- a/src/main/java/org/esipfed/eskg/portalservice/OntoUploader.java
+++ b/src/main/java/org/esipfed/eskg/portalservice/OntoUploader.java
@@ -1,14 +1,43 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you 
+ * may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 package org.esipfed.eskg.portalservice;
 
-import java.io.*;
-import java.lang.Exception;
-import java.lang.String;
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
 import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.ProtocolException;
 import java.net.URL;
+import java.nio.charset.Charset;
 import java.nio.file.Files;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 public class OntoUploader {
+
+  private static final Logger LOG = LoggerFactory.getLogger(OntoUploader.class);
+
   static final String REST_URL = "http://semanticportal.esipfed.org:8080";
+
   private static final String API_KEY = "59f9f403-51f3-417e-9294-f9be7a737f50";
 
   private static String get(String urlToGet) {
@@ -24,75 +53,87 @@ public class OntoUploader {
       conn.setRequestProperty("Authorization", "apikey token=" + API_KEY);
       conn.setRequestProperty("Accept", "application/json");
       rd = new BufferedReader(
-          new InputStreamReader(conn.getInputStream()));
+              new InputStreamReader(conn.getInputStream(), Charset.defaultCharset()));
       while ((line = rd.readLine()) != null) {
         result += line;
       }
       rd.close();
     } catch (Exception e) {
-      e.printStackTrace();
+      LOG.error("Error attempting to execute GET request: {} {}", urlToGet, e);
     }
     return result;
   }
 
   //post text in json format
-  private static String postJSON(String urlToPost, String body) {
-    URL url;
-    HttpURLConnection conn;
+  private static String postJSON(String urlToPost, String body) throws IOException {
+    URL url = null;
+    HttpURLConnection conn = null;
 
     String line;
     String result = "";
+
     try {
       url = new URL(urlToPost);
+    } catch (MalformedURLException e) {
+      LOG.error("Error in URL construction: {} {}", urlToPost ,e);
+    }
+    try {
       conn = (HttpURLConnection) url.openConnection();
-      conn.setDoOutput(true);
-      conn.setDoInput(true);
-      conn.setInstanceFollowRedirects(false);
+    } catch (IOException e) {
+      LOG.error("Error attempting to open URL connection: ", e);
+    }
+    conn.setDoOutput(true);
+    conn.setDoInput(true);
+    conn.setInstanceFollowRedirects(false);
+    try {
       conn.setRequestMethod("POST");
-      conn.setRequestProperty("Authorization", "apikey token=" + API_KEY);
-      conn.setRequestProperty("Accept", "application/json");
-      conn.setRequestProperty("Content-Type", "application/json");
-      conn.setRequestProperty("charset", "utf-8");
-      conn.setUseCaches(false);
+    } catch (ProtocolException e) {
+      LOG.error("Error setting POST request method: ", e);
+    }
+    conn.setRequestProperty("Authorization", "apikey token=" + API_KEY);
+    conn.setRequestProperty("Accept", "application/json");
+    conn.setRequestProperty("Content-Type", "application/json");
+    conn.setRequestProperty("charset", "utf-8");
+    conn.setUseCaches(false);
 
-      DataOutputStream wr = new DataOutputStream(conn.getOutputStream());
-      wr.write(body.getBytes());
-      wr.flush();
-      wr.close();
+    try (DataOutputStream wr = new DataOutputStream(conn.getOutputStream())){
+      wr.write(body.getBytes(Charset.defaultCharset()));
       conn.disconnect();
-
-      InputStream is;
-      boolean error = false;
-      if (conn.getResponseCode() >= 200 && conn.getResponseCode() < 400) {
-        is = conn.getInputStream();
-      } else {
-        error = true;
-        is = conn.getErrorStream();
-      }
-
-      BufferedReader rd = new BufferedReader(new InputStreamReader(is));
-      while ((line = rd.readLine()) != null) {
-        result += line;
-      }
-      rd.close();
-
-      if (error) throw new Exception(result);
-    } catch (Exception e) {
-      e.printStackTrace();
     }
 
+    InputStream is;
+    boolean error = false;
+    if (conn.getResponseCode() >= 200 && conn.getResponseCode() < 400) {
+      is = conn.getInputStream();
+    } else {
+      error = true;
+      is = conn.getErrorStream();
+    }
+
+    BufferedReader rd = new BufferedReader(new InputStreamReader(is, Charset.defaultCharset()));
+    while ((line = rd.readLine()) != null) {
+      result += line;
+    }
+    rd.close();
+
+    if (error)
+      try {
+        throw new Exception(result);
+      } catch (Exception e) {
+        LOG.error("Error retrieving response: ", e);
+      }
     return result;
   }
-  
+
   //post text file
   private static String postFile(String urlToPost, String body, String filePath) {
     URL url;
     HttpURLConnection conn;
-    
+
     String charset = "UTF-8";
     File ontologyFile = new File(filePath);
     String CRLF = "\r\n"; // Line separator required by multipart/form-data.
-    
+
     String line;
     String result = "";
     try {
@@ -107,7 +148,7 @@ public class OntoUploader {
       conn.setRequestProperty("Content-Type", "application/json");
       conn.setRequestProperty("charset", "utf-8");
       conn.setUseCaches(false);
-      
+
       OutputStream output = conn.getOutputStream();
       PrintWriter writer = new PrintWriter(new OutputStreamWriter(output, charset), true);
       writer.append("Content-Disposition: form-data; name=\"textFile\"; filename=\"" + ontologyFile.getName() + "\"").append(CRLF);
@@ -126,28 +167,29 @@ public class OntoUploader {
         is = conn.getErrorStream();
       }
 
-      BufferedReader rd = new BufferedReader(new InputStreamReader(is));
+      BufferedReader rd = new BufferedReader(new InputStreamReader(is, Charset.defaultCharset()));
       while ((line = rd.readLine()) != null) {
         result += line;
       }
       rd.close();
 
-      if (error) throw new Exception(result);
+      if (error)
+        throw new Exception(result);
     } catch (Exception e) {
-      e.printStackTrace();
+      LOG.error("");
     }
 
     return result;
   }
 
   public static void main(String[] args) {
-    String response_get = OntoUploader.get(REST_URL + "/ontologies/GCMD-DIF/submissions");
-    System.out.println(response_get);
-    
+    String getResponse = OntoUploader.get(REST_URL + "/ontologies/GCMD-DIF/submissions");
+    LOG.info(getResponse);
+
     //http://data.bioontology.org/documentation#OntologySubmission
     //need to figure out the format of each parameter in the post body
-    String response_post = OntoUploader.postFile(REST_URL + "/ontologies/GCMD-DIF/submissions", null, "/Users/yjiang/Documents/mudrod_data/ontotest.owl");
-    System.out.println(response_post);
+    String postResponse = OntoUploader.postFile(REST_URL + "/ontologies/GCMD-DIF/submissions", null, "/Users/yjiang/Documents/mudrod_data/ontotest.owl");
+    LOG.info(postResponse);
   }
 
 }

